File: web/src/lib/constants/source.ts
import type { Source } from "@/types/api";

/** The channels a person can pick when entering an application by hand — Application.source, not
 * Job.source. The ATS members of the Source union (Greenhouse, Lever, Workday, ...) are deliberately
 * absent: those describe where a posting was read from and are only ever set by the backend's
 * JobPostingSourceResolver, never chosen in this form. */
export const SOURCES: Source[] = [
  "Manual",
  "LinkedIn",
  "KariyerNet",
  "LinkedInImport",
  "CsvImport",
  "CompanyWebsite",
  "Referral",
  "BrowserExtension",
  "Email",
  "System",
  "Other",
];

File: src/AfterApply.Domain/Common/Source.cs
namespace AfterApply.Domain.Common;

public enum Source
{
    Manual,
    LinkedIn,
    KariyerNet,
    LinkedInImport,
    CsvImport,
    CompanyWebsite,
    Referral,
    BrowserExtension,
    Email,
    System,
    Other,

    // Applicant tracking systems — the page where the application is actually submitted, as
    // opposed to the aggregator that advertised it. Only ever used as Job.Source (provenance of
    // the posting), never as Application.Source (which stays BrowserExtension/Manual/...), and
    // only reachable through JobPostingSourceResolver. Persisted as strings
    // (HasConversion<string>), so adding a member costs no column migration.
    Greenhouse,
    Lever,
    Ashby,
    Workday,
    Workable,
    SmartRecruiters
}
